static/js/utils/string_utils.ts

```diff
@@ -89,6 +89,10 @@ export function getCommonPrefix(words: string[]): string {
  * Given a date string, check that it is in the form YYYY-MM-DD or YYYY-MM or YYYY
  */
 export function isValidDate(date: string): boolean {
+  const dateObj = new Date(date);
+  if (dateObj.toString() === "Invalid Date") {
```

Comment:
This could be platform dependent. Could we instead use:
```
Date.parse(date) !== NaN
```

Reply:
Add a quick test for this too? Since it's a util..

Reply:
done